src/rejectdups-rust/src/lib.rs

```diff
@@ -0,0 +1,500 @@
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    // the voter 
+    let voter_account = next_account_info(accounts_iter)?;
+
+    if !voter_account.is_signer {
+      return Err(ProgramError::MissingRequiredSignature);
+    }
+
+    let expected_check_account_address = create_address_with_seed(voter_account.key, 
+                                                                  "checkvote", 
+                                                                  program_id);
+
+    if expected_check_account_address != *check_account.key {
+        info!("Voter fraud! not the correct check_account.");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    let mut checkdata = check_account.try_borrow_mut_data()?;
+
+    let cac = LittleEndian::read_u32(&checkdata[0..4]);
+
+    if cac!=0 {
+        info!("Voter fraud! You already voted.");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+
+    // Increment vote count of candidate
+
+    let mut data = account.try_borrow_mut_data()?;
+
+    if 1 == instruction_data[0] { 
+        let mut vc = LittleEndian::read_u32(&data[0..4]);
+        vc += 1;
+        LittleEndian::write_u32(&mut data[0..4], vc);
+        info!("Voted for candidate1!");
+        LittleEndian::write_u32(&mut checkdata[0..4], 1);
+    }
+
+    if 2 == instruction_data[0] { 
+        let mut vc = LittleEndian::read_u32(&data[4..8]);
+        vc += 1;
+        LittleEndian::write_u32(&mut data[4..8], vc);
+        info!("Voted for candidate2!");
+        LittleEndian::write_u32(&mut checkdata[0..4], 2);
+    }
+
+    Ok(())
+}
+
+
+// tests
+#[cfg(test)]
+mod test {
+    use super::*;
+    use solana_sdk::clock::Epoch;
+
+ 
+    static SYSTEM_ACCOUNT_PUBKEY_BYTES: &[u8] = &[
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0,
+    ];
+
+    static PROGRAM_ACCOUNT_PUBKEY_BYTES: &[u8] = &[
+        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0,
+    ];
+
+    static DATA_ACCOUNT_PUBKEY_BYTES: &[u8] = &[
+        2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0,
+    ];
+
+    static VOTER_ACCOUNT_PUBKEY_BYTES: &[u8] = &[
+        3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0,
+    ];
+
+    static FALSE_CHECK_ACCOUNT_PUBKEY_BYTES: &[u8] = &[
+        4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0,
+    ];
+
+
+
+    // single vote
+
+    #[test]
+    fn test_sanity() {
+
+        // mock program id 
+
+        let program_id = Pubkey::new( PROGRAM_ACCOUNT_PUBKEY_BYTES );
+
+        // mock contract data account
+
+        let key = Pubkey::new( DATA_ACCOUNT_PUBKEY_BYTES ); // anything 
+        let mut lamports = 0;
+        let mut data = vec![0; 2 * mem::size_of::<u32>()];
```

Comment:
Good spot where `solana_sdk::program_pack::Pack` implemented for the account would simplify this